File: scratch/check_db.js
import Database from 'better-sqlite3';
import path from 'path';

const dbPath = path.resolve('trading_app.db');
const db = new Database(dbPath);

const rows = db.prepare('SELECT id, name FROM strategy_definitions').all();
console.log('--- Strategy Definitions ---');
rows.forEach(row => {
    console.log(`ID: ${row.id} | Name: ${row.name}`);
});

const results = db.prepare('SELECT backtestId, strategyId FROM strategy_backtest_results LIMIT 5').all();
console.log('\n--- Recent Backtest Results ---');
results.forEach(row => {
    console.log(`BT_ID: ${row.backtestId} | StratID: ${row.strategyId}`);
});

File: scratch/check_db.py

import sqlite3
import json

db_path = "trading_app.db"

def check_db():
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    print("--- Tables ---")
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    for table in tables:
        print(f"Table: {table[0]}")
    
    print("\n--- sentiment_scores Schema ---")
    cursor.execute("PRAGMA table_info(sentiment_scores);")
    columns = cursor.fetchall()
    for col in columns:
        print(col)
        
    print("\n--- sentiment_scores Sample Data ---")
    cursor.execute("SELECT * FROM sentiment_scores LIMIT 3;")
    rows = cursor.fetchall()
    for row in rows:
        print(row)
        
    conn.close()

if __name__ == "__main__":
    check_db()
